newlib/libc/include/errno.h

```diff
@@ -43,4 +43,9 @@ typedef int error_t;
 
 #include <sys/errno.h>
 
+#include <sys/_types.h>
+#ifdef __STDC_WANT_LIB_EXT1__
+typedef __errno_t errno_t;
```

Comment:
This also needs to be in stdlib.h -- we'll have to use the same pattern as other typedefs which are required from multiple files:
```
#ifndef _ERRNO_T_DEFINED
typedef __errno_t errno_t;
#define _ERRNO_T_DEFINED
#endif
```
Stick that in errno.h, stdio.h (both tinystdio and legacy stdio), stdlib.h, time.h, string.h, and wchar.h.